bin/logstash

```diff
@@ -58,5 +58,8 @@ if [ "$1" = "-V" ] || [ "$1" = "--version" ]; then
   fi
   echo "logstash $LOGSTASH_VERSION"
 else
-  ruby_exec "${LOGSTASH_HOME}/lib/bootstrap/environment.rb" "logstash/runner.rb" "$@"
+  function join_cp { local IFS=":"; echo "$*"; }
+  jars=(${LOGSTASH_HOME}/logstash-core/lib/jars/*.jar)
```

Comment:
@jordansissel I'd go for 3 here.

1. Is really hacky (and I think even problematic legally in some cases) since we have to flatten all jars (or do some other hack that deals with nested jars) into one which entails removing the checksums from their `META-INF` paths => not something we want to do imo.
2. I don't have a good theoretical argument against this, but the build is already quite complicated isn't it? I'd rather not pile onto that :) Plus I was just able to remove the generated files like logstash_core_jars.rb from the production code paths, I'd rather remove these hacks and not just add a new generated piece of code.

=> `sh` is def more portable than `bash`, let me try your code :) thanks

Reply:
@jordansissel hmm your solution works like a charm, but unfortunately only on newer `sh`. On vanilla OSX with `sh 3.2` the `-n` is not understood by `echo` and the solution will only work with `bash`.
But you know what `bash` must be fine for us ... we are currently calling `jruby/bin/jruby` from our scripts and it's a `bash` script too => we're only compatible if `bash` is available right now anyways.
=> Use your solution but with `bash` to work on systems with older `sh`? :)